src/modlunky2/ui/levels/tab.py

```diff
@@ -3041,6 +3041,54 @@ def save_level(
                 if tilecode[0].split(" ", 1)[0] == "floor_hard":
                     hard_floor_code = tilecode[0].split(" ", 1)[1]
 
+            def write_vanilla_room(x, y, foreground, background):
+                vanilla_setroom_type = (
+                    self.vanilla_setroom_type_for(theme, x, y)
+                    if save_format.include_vanilla_setrooms
+                    else LevelsTab.VanillaSetroomType.NONE
+                )
+                vf = []
+                vb = []
+                vs = []
+                vm = ""
+                dual = (not hard_floor_code) or background != [
+                    hard_floor_code * 10 for _ in range(8)
+                ]
+                if vanilla_setroom_type == LevelsTab.VanillaSetroomType.FRONT:
+                    vf = foreground
+                    vm = "the front layer"
+                elif vanilla_setroom_type == LevelsTab.VanillaSetroomType.BACK:
+                    vf = background
+                    vm = "the back layer"
+                elif vanilla_setroom_type == LevelsTab.VanillaSetroomType.DUAL:
+                    vf = foreground
+                    vm = "both layers"
+                    if dual:
+                        vb = background
+                        vs.append(TemplateSetting.DUAL)
+
+                if vanilla_setroom_type != LevelsTab.VanillaSetroomType.NONE:
+                    template_chunks = [
+                        Chunk(
+                            comment=None,
+                            settings=vs,
+                            foreground=vf,
+                            background=vb,
+                        )
+                    ]
+                    comment_format = (
```

Comment:
I think a formatted string literal would be better
  `comment = f"Auto-generated template to match {vm} of {template_name}."`
(note the `f` before `"`)

Same goes for `name`